### PR TITLE
Fix#11195. com_cache. Button Delete All. Delete Site AND Administrator cache with 1 click

### PR DESCRIPTION
**Pull request by ghost**
Pull Request for Issue https://github.com/joomla/joomla-cms/issues/11195

> "... usability issue that on an update of joomla we tell people to update the cache and right now they are a) mistakenly only cleaning the cache for the site not the admin and b) have to do three extra clicks when it just isnt needed."
#### Summary of Changes
- Button Delete All (caches) deletes cache of Site and Administrator indepentend of Filter client_id.
##### If you have the possibility to use cache handlers other than _File_ then please testtesttest, too. Thx!

EDIT: Be aware that if you use Cache Handler `Cache_Lite` that folders (column "Cache Groups") are not removed but only cache files in it that were created by Cache_Lite! You always will see a "1" in column "Number of Files" after deleting. That's a remaining file `index.html` in each folder. That's not related to this PR.
- administrator/index.php?option=com_config
- \- Activate Cache with Cache Handler _File_
- \- Load some pages in FE and some in BE.
- administrator/index.php?option=com_cache
- \- Via FTP create a folder /administrator/cache/Unbenannt/ (must be writeable/deleteable for Joomla)
- \- See entries with Filter set to _Site_
- \- See entries with Filter set to _Administrator_
- Change Filter to _Site_
- Click on button _Delete All_
- Change Filter to _Administrator_
- You'll see that all folders are still present including folder /Unbenannt/

Apply patch.
- Follow the steps above.
- You'll see that also Administrator folder  /Unbenannt/ has been deleted. (Other folders were newly created immediately after they have been deleted.)
- To see error messages:
- \- Via FTP create a folder /administrator/cache/Unbenannt/ (must be writeable/deleteable for Joomla)
- \- Via FTP create a folder /cache/Unbenannt/ (must be writeable/deleteable for Joomla)
- \- Via FTP create a folder /administrator/cache/Unbenannt1/ (must NOT be writeable/deleteable for Joomla)
- \- Via FTP create a folder /cache/Unbenannt1/ (must NOT be writeable/deleteable for Joomla)
- Click on _Delete All_

---

---

---

---

### CODE REVIEW

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 890747eacb2d060643a64040a5838971be77a1da<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11286">issues.joomla.org/joomla-cms/11286</a>.</sub>

File: administrator/components/com_cache/models/cache.php [hunk 1]
```diff
@@ -165,15 +165,20 @@ public function getData()
 	 *
 	 * @return object
 	 */
-	public function getCache()
+	public function getCache($client_id = null)
```
Comment (andrepereiradasilva) [inline administrator/components/com_cache/models/cache.php]: you need to add this new param to doc block

**Comment by truptikagathara** [comment]
I have tested this item :white_check_mark: successfully on 890747eacb2d060643a64040a5838971be77a1da<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11286">issues.joomla.org/joomla-cms/11286</a>.</sub>

**Comment by brianteeman** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11286">issues.joomla.org/joomla-cms/11286</a>.</sub>

**Pull request merged**